packages/scoutgame/src/projects/updateScoutProject.ts

```diff
@@ -0,0 +1,253 @@
+import { prisma, ScoutProjectMemberRole } from '@charmverse/core/prisma-client';
+import { getContractDeployerAddress } from '@packages/blockchain/getContractDeployerAddress';
+import { v4 } from 'uuid';
+import { verifyMessage } from 'viem';
+
+import { CONTRACT_DEPLOYER_SIGN_MESSAGE } from './constants';
+import type { UpdateScoutProjectFormValues } from './updateScoutProjectSchema';
+
+export async function updateScoutProject(payload: UpdateScoutProjectFormValues, userId: string) {
+  const project = await prisma.scoutProject.findUniqueOrThrow({
+    where: {
+      id: payload.projectId
+    },
+    select: {
+      scoutProjectMembers: {
+        select: {
+          userId: true,
+          deletedAt: true
+        }
+      },
+      scoutProjectDeployers: {
+        select: {
+          address: true
+        }
+      },
+      scoutProjectContracts: {
+        select: {
+          address: true,
+          deletedAt: true
+        }
+      }
+    }
+  });
+
+  const contractTransactionRecord: Record<
+    string,
+    { chainId: number; txHash: string; blockNumber: number; blockTimestamp: number }
+  > = {};
+  const projectMemberIds = project.scoutProjectMembers.map((member) => member.userId);
+  const projectDeployerAddresses = project.scoutProjectDeployers.map((deployer) => deployer.address.toLowerCase());
+  const projectContractAddresses = project.scoutProjectContracts.map((contract) => contract.address.toLowerCase());
+
+  const addedContractAddresses =
+    payload.contracts
+      ?.map((contract) => contract.address.toLowerCase())
+      .filter((address) => !projectContractAddresses.includes(address)) ?? [];
+  const addedDeployerAddresses =
+    payload.deployers
+      ?.map((deployer) => deployer.address.toLowerCase())
+      .filter((address) => !projectDeployerAddresses.includes(address)) ?? [];
+  const addedProjectMemberIds =
+    payload.teamMembers?.map((member) => member.scoutId).filter((scoutId) => !projectMemberIds.includes(scoutId)) ?? [];
+
+  const removedContractAddresses =
+    payload.contracts &&
+    projectContractAddresses.filter(
+      (address) => !payload.contracts!.some((contract) => contract.address.toLowerCase() === address.toLowerCase())
+    );
+  const removedDeployerAddresses =
```

Comment:
i dont see why we should ever delete deployer addresses

Reply:
yeah thats a good point. removed